### PR TITLE
[speechRecognition.py] new mute/unmute funtionality added

### PR DESCRIPTION
**Pull request by rsantos88**
[speechRecognition.py] new mute/unmute funtionality added

### CODE REVIEW

**Comment by PeterBowman** [comment]
Could you please rebase on latest develop? BTW I'm unable to test this, `gst-inspect-1.0 pocketsphinx` prints `No such element or plugin 'pocketsphinx'`. I think the install guides could be improved as several dependencies (e.g. Python packages) are not covered at all.

**Comment by jgvictores** [comment]
I was going to comment on a requirements.txt too. :-)

**Comment by jgvictores** [comment]
> I'm unable to test this, gst-inspect-1.0 pocketsphinx prints No such element or plugin 'pocketsphinx'

@PeterBowman https://github.com/roboticslab-uc3m/speech/blob/74a68d01d268802e4652f8bdbb5397b08f74663a/doc/speech-install.md points to https://github.com/roboticslab-uc3m/installation-guides/blob/master/install-speech-recognition.md where pocketsphinx should be installed. Could you check it put?

**Comment by PeterBowman** [comment]
It looks like the pocketsphinx plugin had been blacklisted by GStreamer.

```shell
$ gst-inspect-1.0 /usr/local/lib/gstreamer-1./libgstpocketsphinx.so
Plugin Details:
  Name                     libgstpocketsphinx.so
  Description              Plugin for blacklisted file
  Filename                 /usr/local/lib/gstreamer-1.0/libgstpocketsphinx.so
  Version                  0.0.0
  License                  BLACKLIST
  Source module            BLACKLIST
  Binary package           BLACKLIST
  Origin URL               BLACKLIST


  0 features:
```

```shell
$ gst-inspect-1.0 -b
Blacklisted files:
  libgstpocketsphinx.so

Total count: 1 blacklisted file
```

Per [this SO question](https://stackoverflow.com/q/27396802), I deleted `$HOME/.cache/gstreamer-1.0/registry.x86_64.bin` and rerun `gst-inspect-1.0`. It's fine, now.

> Could you check it out?

That's what I did in the first place. My PC is still choking on the `sudo modprobe snd_pcm_oss` line (module not found).

**Comment by rsantos88** [comment]
Ok, it was tested in Teo and it works well using `follow-me` demonstration ([see here the changes](https://github.com/roboticslab-uc3m/follow-me/pull/42)) to send mute/unmute commands during the dialogue (more accurately, it mutes microphone when is speaking to avoid listen to himself)
Can we merge this branch already?

File: programs/speechRecognition/speechRecognition.py [hunk 1]
```diff
@@ -112,14 +125,39 @@ def read(self,connection):
...
+            except alsaaudio.ALSAAudioError:
+                print("No such mixer: %s" %sys.stderr)
+                sys.exit(1)
+    
+            channel = alsaaudio.MIXER_CHANNEL_ALL            
+            # mute/unmute
+            if(bottleIn.get(1).asString() ==  "mute"):
+                mixer.setrec(0, channel)
+                bottleOut.addString("microphone muted")
+		print("mic muted")
```
Comment (PeterBowman) [inline programs/speechRecognition/speechRecognition.py]: Tabs instead of spaces.

File: programs/speechRecognition/speechRecognition.py [hunk 2]
```diff
@@ -112,14 +125,39 @@ def read(self,connection):
...
+            elif(bottleIn.get(1).asString() ==  "unmute"):
+                mixer.setrec(1, channel)
+                bottleOut.addString("microphone unmuted")
+		print("mic unmuted")
```
Comment (PeterBowman) [inline programs/speechRecognition/speechRecognition.py]: Tabs instead of spaces.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pyalsaaudio  # for speechRecognition.py
```
Comment (PeterBowman) [inline requirements.txt]: Missing packages for gi/gtk/gobject (might be related). Just noticing that I had to install the `python-gi` apt package where a pip one would be preferable.

File: programs/speechRecognition/speechRecognition.py (hunk 1 above)
Comment (PeterBowman) [reply programs/speechRecognition/speechRecognition.py]: By the way, if you are going to fix those whitespaces, it's worth noting that parens are not necessary around `if` clauses in Python: `if bottleIn.get(1).asString() ==  "mute":`.

**Comment by PeterBowman** [comment]
Wait, I did not suggest to replace `print('hey')` with `print 'hey'`, just `if` clauses... I guess this won't work now on Python 3. Also, there is still some room to improve readability: `not bottleIn.read(connection)` instead of `not(bottleIn.read(connection))`.

**Comment by rsantos88** [comment]
It works with Python 2.7.6 but feel free to change whatever you want

**Comment by rsantos88** [comment]
> Wait, I did not suggest to replace print('hey') with print 'hey', just if clauses... I guess this won't work now on Python 3. Also, there is still some room to improve readability: not bottleIn.read(connection) instead of not(bottleIn.read(connection)).

Changes done as I've talked with @jgvictores . I hope everything's ok.

File: requirements.txt (hunk 3 above)
Comment (jgvictores) [reply requirements.txt]: Now commented at https://github.com/roboticslab-uc3m/speech/blob/41336b99e01fad4c8ad9244dd2c7fe18fc9182ee/doc/speech-install.md#troubleshooting-installation (was not required for me, on a clean Xenial + pip2)

**Comment by jgvictores** [comment]
> It looks like the pocketsphinx plugin had been blacklisted by GStreamer. 

Re https://github.com/roboticslab-uc3m/speech/pull/18#issuecomment-380416736 : Not sure what was happening there, but order is important for `gst-inspect-1.0 pocketsphinx` ([verification](https://github.com/roboticslab-uc3m/installation-guides/blob/0820e41e13009904c70798219a59a5b38dbd4da4/install-gstreamer-pocketsphinx.md#verify-gstreamer-with-pocketsphinx)). I've documented the importance of first `gstreamer` then `pocketsphinx` at at https://github.com/roboticslab-uc3m/installation-guides/blob/0820e41e13009904c70798219a59a5b38dbd4da4/install-gstreamer-pocketsphinx.md#troubleshooting-gstreamer-with-pocketsphinx

**Pull request merged**